File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDODateTimeParamModel.kt
//
//  IDODateTimeParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.annotations.SerializedName

import com.google.gson.Gson
import com.google.gson.GsonBuilder

///

/// 设置时间

open class IDODateTimeParamModel(
    year: Int,
    monuth: Int,
    day: Int,
    hour: Int,
    minute: Int,
    second: Int,
    week: Int,
    timeZone: Int
) : IDOBaseModel {


    @SerializedName("year")
    var year: Int = year

    @SerializedName("monuth")
    var monuth: Int = monuth

    @SerializedName("day")
    var day: Int = day

    @SerializedName("hour")
    var hour: Int = hour

    @SerializedName("minute")
    var minute: Int = minute

    @SerializedName("second")
    var second: Int = second

    /// Week: 0-6 for Monday to Sunday
    @SerializedName("week")
    var week: Int = week

    /// Timezone in a 24-hour format: 0-12 for East ,13-24 for West
    @SerializedName("time_zone")
    var timeZone: Int = timeZone


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDODateTimeParamModel.swift
//
//  IDODateTimeParamModel.swift
//  alexa_channel
//
//  Created by hc on 2023/10/11.
//

import Foundation

/// 设置时间
@objcMembers
public class IDODateTimeParamModel: NSObject, IDOBaseModel {
    
    public var year, monuth, day, hour: Int
    public var minute, second: Int
    /// Week: 0-6 for Monday to Sunday
    public var week: Int
    /// Timezone in a 24-hour format: 0-12 for East ,13-24 for West
    public var timeZone: Int

    enum CodingKeys: String, CodingKey {
        case year, monuth, day, hour, minute, second, week
        case timeZone = "time_zone"
    }

    public init(year: Int, monuth: Int, day: Int, hour: Int, minute: Int, second: Int, week: Int, timeZone: Int) {
        self.year = year
        self.monuth = monuth
        self.day = day
        self.hour = hour
        self.minute = minute
        self.second = second
        self.week = week
        self.timeZone = timeZone
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
    
}
